hyperspy/tests/io/test_hdf5.py

```diff
@@ -229,3 +230,33 @@ def test_rgba16():
         "npy_files",
         "test_rgba16.npy"))
     nt.assert_true((s.data == data).all())
+
+
+class TestLoadingOOMReadOnly:
+
+    def setUp(self):
+        s = Signal(np.empty((5, 5, 5)))
+        s.save('tmp.hdf5')
+        self.shape = (10000, 10000, 100)
+        del s
+        f = h5py.File('tmp.hdf5', model='r+')
+        s = f['Experiments/__unnamed__']
+        del s['data']
+        s.create_dataset(
+            'data',
+            shape=self.shape,
+            dtype='float64',
+            chunks=True)
+        f.close()
+
+    @nt.raises(MemoryError)
```

Comment:
...I __could__ actually just check what's the available memory and then create the data somewhat bigger (`psutil` is wonderful, in case you did not know ;) ).. would probably only be marginally slower and we'd be safe for eternity :)